src/cobra/util/solver.py

```diff
@@ -275,86 +300,111 @@ def choose_solver(model, solver=None, qp=False):
     solver object.
 
     Parameters
     ----------
-    model : a cobra model
+    model : cobra.Model
        The model from which to remove the variables and constraints.
-    what : list or tuple of optlang variables or constraints.
-       The variables or constraints to remove from the model. Must be of
-       class `model.problem.Variable` or
-       `model.problem.Constraint`.
-    """
-    context = get_context(model)
+    what : list or tuple of optlang.interface.Variable or
+           optlang.interface.Constraint
+       The variables and constraints to remove from the model.
 
+    """
     model.solver.remove(what)
+
+    context = get_context(model)
     if context:
         context(partial(model.solver.add, what))
 
 
 def add_absolute_expression(
-    model, expression, name="abs_var", ub=None, difference=0, add=True
-):
+    model: "Model",
+    expression: str,
+    name: str = "abs_var",
+    ub: Optional[float] = None,
+    difference: Optional[float] = 0.0,
+    add: bool = True,
+) -> NamedTuple:
     """Add the absolute value of an expression to the model.
 
-    Also defines a variable for the absolute value that can be used in other
-    objectives or constraints.
+    Also defines a variable for the absolute value that can be used in
+    other objectives or constraints.
 
     Parameters
     ----------
-    model : a cobra model
+    model : cobra.Model
        The model to which to add the absolute expression.
-    expression : A sympy expression
-       Must be a valid expression within the Model's solver object. The
-       absolute value is applied automatically on the expression.
-    name : string
-       The name of the newly created variable.
-    ub : positive float
-       The upper bound for the variable.
-    difference : positive float
-        The difference between the expression and the variable.
-    add : bool
-        Whether to add the variable to the model at once.
+    expression : str
+       Must be a valid symbolic expression within the model's solver object.
+       The absolute value is applied automatically on the expression.
+    name : str, optional
+       The name of the newly created variable (default "abs_var").
+    ub : positive float, optional
+       The upper bound for the variable (default None).
+    difference : positive float, optional
+        The difference between the expression and the variable
+        (default 0.0).
+    add : bool, optional
+        Whether to add the variable to the model at once (default True).
 
     Returns
     -------
-    namedtuple
+    NamedTuple
         A named tuple with variable and two constraints (upper_constraint,
         lower_constraint) describing the new variable and the constraints
         that assign the absolute value of the expression to it.
+
     """
-    Components = namedtuple(
-        "Components", ["variable", "upper_constraint", "lower_constraint"]
+    Components = NamedTuple(
```

Comment:
Can you define `Components` at module level, please, rather than within the function and then set the return type of this function to `-> Components`.

Reply:
Sure no problem.